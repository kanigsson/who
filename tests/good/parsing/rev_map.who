type list['a] =
| Nil
| Cons of 'a * 'a list

let rev_map (_ : int) =
  {}
  let rec aux (_ : int) : int,{} = {} aux 1 {} in 5
  {}
