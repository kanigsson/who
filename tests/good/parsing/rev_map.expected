type list ['a||] = | Nil | Cons of a* list[a] 
let
  rev_map  = (fun (anon:int) -> {(fun (cur:< >) -> True)}
             let
             rec(int ->{  +  } int) aux =
                                         (fun (anon1:int) ->
                                         {(fun (cur:< >) -> True)} aux 1
                                         {(fun (old:< >) ->
                                          (fun (cur:< >) ->
                                          (fun (anon2:int) -> True)))})
             in 5
             {(fun (old:< >) -> (fun (cur:< >) -> (fun (anon1:int) -> True)))}) 
